silkworm/silkrpc/storage/remote_chain_storage.cpp

```diff
@@ -54,8 +54,13 @@ Task<bool> RemoteChainStorage::read_block(const Hash& hash, BlockNum number, sil
 }
 
 Task<bool> RemoteChainStorage::read_block(const Hash& hash, silkworm::Block& block) const {
-    const auto number = co_await core::rawdb::read_header_number(reader_, hash);
-    co_return co_await backend_->get_block(number, hash.bytes, /*.read_senders=*/false, block);
+    uint64_t block_number{0};
+    try {
+        block_number = co_await core::rawdb::read_header_number(reader_, hash);
+    } catch (const std::invalid_argument& iv) {
+        co_return false;
+    }
```

Comment:
As discussed, we should still rely on `std::invalid_argument` raised by low-level functions here without catching it at this level. Changing this behaviour consistently in the internal APIs is IMHO has broader scope than this PR.

Reply:
ok